### PR TITLE
Add trim(), split_unit() and split_string() Lua helper functions

### PR DESCRIPTION
**Pull request by joto**

### CODE REVIEW

File: src/init.lua [hunk 1]
```diff
@@ -116,6 +116,36 @@ function osm2pgsql.make_clean_tags_func(keys)
...
 end
 
+-- from http://lua-users.org/wiki/StringTrim
+function osm2pgsql.trim(str)
+    local from = str:match("^%s*()")
+    return from > #str and "" or str:match(".*%S", from)
+end
+
+function osm2pgsql.split_unit(str, default_unit)
+    local val, unit = string.match(str, "^([0-9.]+) ?(%a*)$")
```
Comment (lonvia) [inline src/init.lua:126]: Do we want to allow negative values?

Comment (joto) [reply src/init.lua:126]: I had thought about that and don't see where this would be useful. `width`, `maxspeed` can only be positive. Maybe for `ele`?

Comment (joto) [reply src/init.lua]: Its a low-level function that shoulde decide what the correct range of values is, so I have changed it to allow negative values.

**Comment by joto** [comment]
PR has now an additional commit adding checks for nil on several Lua functions.

**Comment by pnorman** [comment]
What is the value to adding these as helper functions?

**Comment by joto** [comment]
See discussion in #1079 . These are just some pretty generic and useful functions and you got to start somewhere.

**Pull request merged**